h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java

```diff
@@ -828,6 +828,248 @@ private Vec createRandomBinaryWeightsVec(final long len, final int randomSeed) {
+        crossValidationSubset = crossValidationSubset.deepCopy("trainingFrameSubset");
+        DKV.put(crossValidationSubset);
+        parms._train = crossValidationSubset._key;
+        XGBoostModel crossValidationModel = null;
+        try {
+          crossValidationModel = new hex.tree.xgboost.XGBoost(parms).trainModel().get();
+
+          ModelMetricsRegression crossValidationModelMetrics = (ModelMetricsRegression) crossValidationModel._output._training_metrics;
+          ModelMetricsRegression originalCrossValidationMetrics = (ModelMetricsRegression) ((XGBoostModel) originalCrossValidationModels[i].get())._output._training_metrics;
+
+          assertEquals(originalCrossValidationMetrics.rmse(), crossValidationModelMetrics.rmse(), 1e-20);
+          assertEquals(originalCrossValidationMetrics.mean_residual_deviance(), crossValidationModelMetrics._mean_residual_deviance, 1e-6); // RMSE one line above, testing correct assignment, even if the value is derived
+          assertEquals(originalCrossValidationMetrics._root_mean_squared_log_error, crossValidationModelMetrics._root_mean_squared_log_error, 1e-20);
+          assertEquals(originalCrossValidationMetrics._mean_absolute_error, crossValidationModelMetrics._mean_absolute_error, 1e-6);
+          assertEquals(originalCrossValidationMetrics._sigma, crossValidationModelMetrics._sigma, 1e-20);
+          assertEquals(originalCrossValidationMetrics._nobs, crossValidationModelMetrics._nobs, 1e-20);
+        } finally {
+          if (crossValidationModel != null) {
+            crossValidationModel.deleteCrossValidationPreds();
+            crossValidationModel.deleteCrossValidationModels();
+            crossValidationModel.remove();
+          }
+        }
+
+      }
+
+    } finally {
+      Scope.exit();
+      if (prostateFrame != null) prostateFrame.remove();
+      if (model != null) {
+        model.deleteCrossValidationModels();
+        model.deleteCrossValidationPreds();
+        model.remove();
+      }
+      if (crossValidationSubset != null) {
+        crossValidationSubset.remove();
+      }
+    }
+
+  }
+
+  /**
+   * Creates a {@link Vec} representing a randomly-generated fold column.
+   *
+   * @param len        Length of the resulting fold column vector
+   * @param randomSeed Seed for the random generator (for reproducibility)
+   * @param nfolds     Number of folds. Basically number of categories in the fold column.
+   * @return An instance of {@link Vec} with integers ranging from 0 to nfoldsm representing the number of categories.
+   */
+  private Vec createRandomFoldColumn(final long len, final int nfolds, final int randomSeed) {
+    if (nfolds < 2) throw new IllegalArgumentException("Number of folds must be greater than or equal to 2.");
+    final Vec weightsVec = Vec.makeZero(len, Vec.T_NUM);
+    final Random random = new Random(randomSeed);
+    for (int i = 0; i < weightsVec.length(); i++) {
```

Comment:
It doesn't matter in the test but we should perpetuate the use of the right APIs in tests as well, please use Vec Writer:

      try (Vec.Writer vw = vec.open()) {
        // for loop with set calls on vw
      }